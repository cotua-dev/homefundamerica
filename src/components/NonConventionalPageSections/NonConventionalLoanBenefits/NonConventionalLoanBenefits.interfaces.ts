export interface FetchNonConventionalLoanBenefitsImageQuery {
    nonConventionalLoanBenefitsImage: {
        publicURL: string;
    };
}
