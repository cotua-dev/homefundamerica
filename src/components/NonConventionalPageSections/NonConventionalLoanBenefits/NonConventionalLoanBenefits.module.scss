section#nonConventionalLoanBenefitsSection {
    height: 60vh;

    @media screen and (max-width: 950px) {
        height: 100%;
    }
}
