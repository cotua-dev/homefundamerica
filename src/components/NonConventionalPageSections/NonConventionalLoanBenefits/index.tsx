import React, {
    FunctionComponent,
    ReactElement,
} from 'react';
import {
    useStaticQuery,
    graphql,
} from 'gatsby';
import { FetchNonConventionalLoanBenefitsImageQuery } from './NonConventionalLoanBenefits.interfaces';
import { nonConventionalLoanBenefitsSection } from './NonConventionalLoanBenefits.module.scss';
import * as sharedStyles from '../../sharedStyles/shared.module.scss';

export const NonConventionalLoanBenefits: FunctionComponent = (): ReactElement => {
    const queryResult = useStaticQuery<FetchNonConventionalLoanBenefitsImageQuery>(graphql`
        query {
            nonConventionalLoanBenefitsImage: file(name: {eq: "AdobeStock_300096295"}, extension: {eq: "jpg"}) {
                publicURL
            }
        }
    `);

    return (
        <section id={nonConventionalLoanBenefitsSection} className={sharedStyles.loanNegative}>
            <div className={sharedStyles.imageWrapper}>
                <img
                    src={queryResult?.nonConventionalLoanBenefitsImage.publicURL}
                    alt="Non conventional loan benefits image"
                />
            </div>
            <div className={sharedStyles.contentWrapper}>
                <h2>{`Non-Conventional Loan Benefits`}</h2>
                <p>
                    {`
                        A Non-Conventional Loan can be an option when your loan
                        amount exceeds the conventional loan limit, $510,400
                        for most U.S. counties.
                    `}
                </p>
                <strong>{`Potential benefits include:`}</strong>
                <ul>
                    <li>{`Jumbo and Super Jumbo loan amounts of up to $25 million or more`}</li>
                    <li>{`Flexibility to choose either a fixed-rate or an adjustable-rate mortgage (ARM)`}</li>
                    <li>{`Personalization for complex financial situations`}</li>
                    <li>{`Financing for non-owner occupied properties`}</li>
                </ul>
            </div>
        </section>
    );
}
