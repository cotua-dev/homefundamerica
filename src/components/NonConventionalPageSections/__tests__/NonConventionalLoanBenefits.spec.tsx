import React from 'react';
import {
    render,
    RenderResult,
} from '@testing-library/react';
import { NonConventionalLoanBenefits } from '../NonConventionalLoanBenefits';

describe('NonConventionalLoanBenefits Component', (): void => {
    test('renders', (): void => {
        const { container }: RenderResult = render(<NonConventionalLoanBenefits/>);
        expect(container.firstChild).toMatchSnapshot();
    });
});
