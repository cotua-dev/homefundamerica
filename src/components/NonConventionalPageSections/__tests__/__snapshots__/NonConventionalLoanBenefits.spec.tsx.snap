// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NonConventionalLoanBenefits Component renders 1`] = `
<section
  id="nonConventionalLoanBenefitsSection"
>
  <div>
    <img
      alt="Non conventional loan benefits image"
    />
  </div>
  <div>
    <h2>
      Non-Conventional Loan Benefits
    </h2>
    <p>
      
                        A Non-Conventional Loan can be an option when your loan
                        amount exceeds the conventional loan limit, $510,400
                        for most U.S. counties.
                    
    </p>
    <strong>
      Potential benefits include:
    </strong>
    <ul>
      <li>
        Jumbo and Super Jumbo loan amounts of up to $25 million or more
      </li>
      <li>
        Flexibility to choose either a fixed-rate or an adjustable-rate mortgage (ARM)
      </li>
      <li>
        Personalization for complex financial situations
      </li>
      <li>
        Financing for non-owner occupied properties
      </li>
    </ul>
  </div>
</section>
`;
